### PR TITLE
feat: add datasets

### PR DESCRIPTION
**Pull request by jolars**

### CODE REVIEW

File: utils/utils.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    from sklearn.preprocessing import MaxAbsScaler, StandardScaler
+
+
+def preprocess_data(X, y=None, remove_zerovar=True, standardize=True):
+    if remove_zerovar:
+        X = VarianceThreshold().fit_transform(X)
+
+    if standardize:
+        if sparse.issparse(X):
+            X = MaxAbsScaler().fit_transform(X).tocsc()
```
Comment (josephsalmon) [inline utils/utils.py:15]: I understand the interest for the sparse case but then why not using the l2 norm as a normalisation?
For me we could do something like:

```
if sparse.issparse(X)==True:
    X = StandardScaler(with_mean=False).fit_transform(X).tocsc()
else:
    X = StandardScaler(with_mean=True).fit_transform(X)
```
that would be more consistent. WDYT?
Then I agree we could offer other options (like l_infy, l1, etc. for instance) later on.

Comment (jolars) [reply utils/utils.py:15]: Regarding our conversation, I think `StandardScaler()` still uses the standard deviation, even if `with_mean=False`. So I'm not sure it's such a good idea. By the way, sci-kit learn recommends this approach (mean+std for dense, maxabs for sparse): https://scikit-learn.org/stable/modules/preprocessing.html#scaling-sparse-data.

Comment (josephsalmon) [reply utils/utils.py:15]: Good catch on `StandardScaler`. It does not do what I expected: with `with_mean=False` I was expecting it would compute l2 norm and not std for normalisation, that's a bit counterintuitive to me.

Thanks for pointing the `sklearn` doc, interesting!
Let's follow their way then, so I will validate you PR.

Yet I find it weird to use l_infty normalisation (so data are in the [-1,1] hypercube), when everybody is used to l2 normalization (std is doing that in a way).

**Review by josephsalmon — APPROVED** [review]
